api/hiveextension/v1beta1/agentclusterinstall_types.go

```diff
@@ -155,8 +155,22 @@ type AgentClusterInstallSpec struct {
 	// DiskEncryption is the configuration to enable/disable disk encryption for cluster nodes.
 	// +optional
 	DiskEncryption *DiskEncryption `json:"diskEncryption,omitempty"`
+
+	// PlatformType is the type of the platform
+	// +optional
+	PlatformType *PlatformType `json:"platformType,omitempty"`
```

Comment:
I would model this like we have in the swagger:
Platform:
   PlatformType

Reply:
Also, this should affect day1 clusters as well - the controller code should handle that.

Reply:
Fixed here: https://github.com/openshift/assisted-service/pull/2975/

Reply:
The comment isn't addressed.  Here you should make a new platform struct that has a string inside it, and the controller shouldn't only set the platform for day2 but should also set it for day1.

Reply:
Will do.